lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -134,34 +136,45 @@ OpaqueResultTypeRequest::evaluate(Evaluator &evaluator,
       return nullptr;
     }
   } else {
-    opaqueReprs = repr->collectOpaqueReturnTypeReprs();
+      if (ctx.LangOpts.hasFeature(Feature::ImplicitSome)) {
+          opaqueReprs = collectTypeReprs(repr);
+      } else {
+          opaqueReprs = collectOpaqueReturnTypeReprs(repr);
+      }
     SmallVector<GenericTypeParamType *, 2> genericParamTypes;
     SmallVector<Requirement, 2> requirements;
     for (unsigned i = 0; i < opaqueReprs.size(); ++i) {
       auto *currentRepr = opaqueReprs[i];
+        
+      if (isa<IdentTypeRepr>(currentRepr)){
+        if(!isProtocol(evaluator, currentRepr, ctx, dc))
```

Comment:
You mentioned this to me offline but I'll record it here 🙂 - if we check `isProtocol` inside of `collectOpaqueReturnTypeReprs`, we won't need to duplicate this code here and inside `createOpaqueParameterGenericParams`.

Reply:
yes! updated 💯